datastore/elastic_search/transfer.py

```diff
@@ -388,14 +389,52 @@ def transfer_data_internal(es_url, index_to_backup, backup_index):
-                                         params={"refresh": "true", "wait_for_completion": "true"})
-        ner_logger.debug('End post request made with es_url ')
-        if reindex_response.status_code != 200:
-            message = "transfer from " + index_to_backup + "to " + backup_index + " failed"
-            raise InternalBackupException(message)
+        number_of_replicas = index_to_backup_config["settings"]["index"].get("number_of_replicas")
+        replicas_disabled = False
+
+        def update_index_settings(index_url, settings_object={}):
+            resp = requests.put(f'{index_url}/_settings', json=settings_object)
+            if resp.status_code == 200:
+                ner_logger.debug(f'transfer_data_internal, Index settings updated')
+            else:
+                ner_logger.debug(f'transfer_data_internal, Failed to update index settings')
+                message = f"Failed to update index settings for {index_url}"
+                raise InternalBackupException(message)
+
+        exception_messages = []
+        # if replica disabling flag is set, then disable replica and re-enable it once reindexing is done
+        if disable_replicas:
+            try:
+                index_settings = {'index': {'number_of_replicas': '0'}}
+                update_index_settings(backup_index_url, index_settings)
+                replicas_disabled = True
+            except InternalBackupException:
+                exception_messages.append('Failed to Disable Replica')
+                disable_replicas = False
+        try:
+            ner_logger.debug('Start post request made with es_url reindex'
+                             'es_url: %s' % (es_url))
+            reindex_response = requests.post('{es_url}/_reindex'.format(**{'es_url': es_url}), json=final_request_dict,
+                                             params={"refresh": "true", "wait_for_completion": "true"})
+            ner_logger.debug('End post request made with es_url ')
+            if reindex_response.status_code != 200:
+                message = "transfer from " + index_to_backup + "to " + backup_index + " failed"
+                exception_messages.append(message)
+        except Exception as exp:
+            ner_logger.debug(f'transfer_data_internal, Error while reindexing : {str(exp)}')
+        finally:
+            # if disable_replicas flag was provided, then check it and re-enable replicas
+            try:
+                if disable_replicas and replicas_disabled:
+                    index_settings = {'index': {'number_of_replicas': number_of_replicas}}
+                    update_index_settings(backup_index_url, index_settings)
+            except InternalBackupException:
+                exception_messages.append('Failed to Re-Enable replicas')
+            finally:
+                if len(exception_messages) > 0:
+                    messages = str(exception_messages)
+                    raise InternalBackupException(messages)
```

Comment:
Add a log message with type exception here unless the caller reading this exception `InternalBackupException` is already taking care of it.

Reply:
Caller function is handling this exception but not logging it.
Will add debug level log to help in case of debugging. and raise the exception as well